docs/user-guide/deployments.md

```diff
@@ -0,0 +1,284 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/user-guide/deployments.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Deployment
+
+**Table of Contents**
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Deployment](#deployment)
+  - [What is a _Deployment_?](#what-is-a-deployment)
+  - [Enabling Deployments on kubernetes cluster](#enabling-deployments-on-kubernetes-cluster)
+  - [Creating a Deployment](#creating-a-deployment)
+  - [Updating a Deployment](#updating-a-deployment)
+  - [Alternative to Deployments](#alternative-to-deployments)
+    - [kubectl rolling update](#kubectl-rolling-update)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## What is a _Deployment_?
+
+A _Deployment_ provides declarative update for Pods and ReplicationControllers.
+Users can define deployments to create new resources, or replace existing ones
+by new ones.
+
+A typical use case is:
+* Create a deployment to bring up a replication controller and pods.
+* Later, update that deployment to recreate the pods (for ex: to use a new image).
```

Comment:
The update can be done in 2 ways: Recreate and rolling update. I have mentioned them in the Deployment strategies section